### DISCUSSION TITLE
lolMiner 1.37

### DISCUSSION THREAD

**Discussion by Lolliedieb** [Lolliedieb/lolMiner-releases#873] [General]
# lolMiner 1.37
(Note: The windows version will be delayed a bit - we need to make sure first LHR unlocking is working there as expected)

### Feature Changes

- Improved Ethash performance on Turing based graphic cards (GTX 16 series, RTX 20 series, lower tier CMP cards) by about 0.4 to 0.7%.
- Improvement of Ethash performance (up to 1%) and reduction of stale share rate for Maxwell and Pascal (GTX 10) based GPUs.
- Changes LHR tuning algorithm to minimize the number of locks & time for finding a stable value: The tuning might be improved by the miner later once stable for long enough time.
- Improved LHR performance for 3060 V1 (GA106-300-A1) on older drivers (460.39 and earlier)
- Added Ergo kernels for RX 6700XT and RX 6600 (XT) on newer AMD drivers.
- Ethereum stratum code will now print the pool difficulty in better human readable number
- The miner will now print the ip of the connected pool - to be able to detect e.g. faulty DNS entries
- Added parameters --dualtls and --dualworker to toggle TLS and the worker name for the dual connection. To be used as with the parameters for the standard connection. 
- **--statsformat** can now distinguish between the number of LHR locks "lhrlks" and the current --lhrtune parameter "lhrparam" 
- Added a parameter **--vstats <number>** to modify the terminal width in number of characters to overwrite the automatic detection, which sometimes does not work with some terminal emulators like putty.
- Added a parameter **--hstats <optional number>**. Given this parameter will switch the statistics to a way such that the statistics write a horizontal line per GPU instead of the default vertical (see screenshot below). When the automatic read terminal width is reached, the remaining entries will be printed to a 2nd set of lines. The vertical statistics can be combined with the usual --statsformat patterns to customize the displayed values per card. The optional number overwrites the the automatic screen width detection. To fit all values of **--statsformat extended** into one line we need **--hstats 150** or higher (recommended setting this value manually when accessing the rig via terminal emulator).

### Fixes 
- Fixed 3060 V1 (GA106-300-A1) internal parameters for drivers 460.39 and below, stabilizing the hash rate (they pretty much did not work stably unlocked in 1.36)
- Fixed a bug causing a wavy hash-rate report in some FHR rigs
- Fixed a bug causing **--mode a** kernels to be defect for Ampere and Turing GPUs in 1.36(a) releases
- Fixed a bug causing a segmentation fault when trying to mine EXCC.

### Preview of new --hstats format 
![grafik](https://user-images.githubusercontent.com/40234439/143767761-21e0ed9e-ee90-4bd9-a917-1ab65b010333.png)

<hr /><em>This discussion was created from the release <a href='https://github.com/Lolliedieb/lolMiner-releases/releases/tag/1.37'>lolMiner 1.37</a>.</em>

**Comment by oldloray** [comment]
could you please make a binary with glib 2.23? Using Linux Mint 18.3 so my last binary which works is 1.34a. Sooner or later i have to update, but a last binary for glib 2.23 would be great to win time ;) Thanks a lot!

**Reply by spajdrEX** [reply]
Hi, how it looks with win version?

**Reply by Lolliedieb** [reply]
Oh ... hmm well my build system at the moment using the about 4.5 years old glib 2.25 ....
I need to think about if there is an easy way to go for an older version without using an ages old compiler (that was different with 1.34a ... that was a kind of emergency release and needed to be done rather quickly on an old laptop of mine ^^ )

**Comment by oldloray** [comment]


**Reply by spajdrEX** [reply]
That was question for the devs, not you :-)

**Comment by GizDS** [comment]
Values need to be specified in --statsformat to see the data of the dual mining:
Speed2 : Speed at second algo in Mhs
PoolHr2 : Pool Average at second algo based submitted shares since startup
Shares2 : Accepted / Stale and invalid shares at second algo since startup
BestShare2 : Difficulty of the most valuated share at second algo since startup
HrPerWatt2 : Efficiency of Mhs you get for 1 Watts usage at second algo
LhrLks : Number of LHR locks since start
LhrParam : Current value of --lhrtune parameter
